### PR TITLE
Done.

### PR DESCRIPTION
**Pull request by yushi1007**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,88 +1,135 @@
...
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
+// const counter = 2
+
+// var counter = 1
+// var counter = 2
+
+let counter = 1
```
Comment (lindsayMecher) [inline index.js:9]: Nice :100: `let` allows us to reassign our variables

File: index.js [hunk 2]
```diff
@@ -1,88 +1,135 @@
...
+counter = 2
```
Comment (lindsayMecher) [inline index.js:10]: Great! The key here was using `const` because it doesn't allow us to reassign our variable

File: index.js [hunk 3]
```diff
@@ -1,88 +1,135 @@
...
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
+const name = "Yu"
```
Comment (lindsayMecher) [inline index.js:24]: Great! The key here was using `const` because it doesn't allow us to reassign our variable

File: index.js [hunk 4]
```diff
@@ -1,88 +1,135 @@
...
+function drinkWater(currentThirstLevel) {
+  console.log("Man, I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+
+console.log("%cQuestion 3", "color: red") 
+
+console.log(drinkWater(12))
```
Comment (lindsayMecher) [inline index.js:48]: Great! You can further simplify to:

```
function drinkWater(thirstLevel) {
  console.log("Man I sure am thirsty");
  console.log("Ahh that hits the spot");
  return thirstLevel -= 1;
}
```

File: index.js [hunk 5]
```diff
@@ -1,88 +1,135 @@
...
+    return "same same"
+  }
+  else{
+
+    return "same same (but different)"       
+}
+}
+console.log("%cQuestion 4", "color: red") 
+
+console.log(sameSameButDifferent(5, 5))
```
Comment (lindsayMecher) [inline index.js:71]: Very close but not quite. We will need an if, else if and else to cover all the conditionals

```
function sameSameButDifferent(num, maybeNum) {
  if (num === maybeNum) {
    console.log("same same")
  } else if (num == maybeNum) {
    console.log("same same but different")
  } else {
    console.log("different")
  }
}
```

File: index.js [hunk 6]
```diff
@@ -1,88 +1,135 @@
...
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+
+function updateGrade(student, grade) {
+
+student.grade = grade
```
Comment (lindsayMecher) [inline index.js:91]: great job!

File: index.js [hunk 7]
```diff
@@ -1,88 +1,135 @@
...
+console.log("%cQuestion 6", "color: red")
+
+function printNameAndPhones(users) {
+  users.forEach(function(currentVal){
+    console.log(`Name: ${currentVal.name}`)
+    console.log(`Cell:${currentVal.phones.cell}`)
+    console.log(`Office:${currentVal.phones.office}`)
+  })
+}
+
```
Comment (lindsayMecher) [inline index.js:116]: great job!!

File: index.js [hunk 8]
```diff
@@ -100,23 +147,33 @@ function myMap(array, callback) {
...
 
-// console.log(myMap(["Raffy", "Chase"], greet)) 
+function greet(name) {
+  return `Hi, ${name}!`
+}
+
+console.log(myMap(["Raffy", "Chase"], greet)) 
 // // => ["Hi, Raffy!", "Hi, Chase!"]
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
```
Comment (lindsayMecher) [inline index.js:176]: Very good, you are just missing the myMap function

```
function myMap(arr, fn) {
  return arr.map(function(item){
    return fn(item);
  });
}
```

File: index.js [hunk 9]
```diff
@@ -141,44 +198,63 @@ const line = []
...
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let ticketNumber = 0
+
+function takeATicketNumber(line){
+ticketNumber ++
+line.push(ticketNumber)
+return `Welcome. You are ticket number ${ticketNumber}`
+}
```
Comment (lindsayMecher) [inline index.js:209]: awesome!

File: index.js [hunk 10]
```diff
@@ -141,44 +198,63 @@ const line = []
...
+  let ticketNumber = 0
 
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 2`
+  return function (line) {
+    ticketNumber += 1
+    line.push(ticketNumber)
+    return `Welcome, You are ticket ${ticketNumber}`
+  }
+}
```
Comment (lindsayMecher) [inline index.js:240]: great work!

**Review by lindsayMecher — COMMENTED** [review]
Great work, Yu!

**Comment by yushi1007** [comment]
> Great work, Yu!

Thank you!

**Pull request closed without merging**